presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -167,9 +173,34 @@ protected Boolean visitAtTimeZone(AtTimeZone node, Void context)
         @Override
         protected Boolean visitSubqueryExpression(SubqueryExpression node, Void context)
         {
+            /*
+             * Walk the subquery and check usages of our fields (the fields defined in our scope) that are not part of GROUP BY.
+             */
+            AstUtils.preOrder(node)
+                    .filter(Predicates.isInstanceOfAny(DereferenceExpression.class, QualifiedNameReference.class))
+                    .map(Expression.class::cast)
+                    .forEach(this::analyzeSubqueryReference);
+
             return true;
         }
 
+        private void analyzeSubqueryReference(Expression referenceExpression)
+        {
+            ResolvedField resolvedField = analysis.getScope(referenceExpression)
+                    .tryResolveField(referenceExpression)
+                    // Unresolvable fields should be found and reported earlier.
```

Comment:
`ExpressionAnalyzer` does that, but I'm reluctant in putting this information here. My intention is that: _if_ after some code changes the exception is being raised in the next line _then_ either we need proper check-n-report here _or_ think more about flows.

In short, the exception servers only as an explanation why I don't do proper reporting here.